lightning_client.go

```diff
@@ -3469,3 +3475,32 @@ func (s *lightningClient) QueryRoutes(ctx context.Context,
 		TotalAmtMsat:  lnwire.MilliSatoshi(route.TotalAmtMsat),
 	}, nil
 }
+
+func (s *lightningClient) CheckMacaroonPermissions(ctx context.Context,
+	macaroon []byte, permissions []MacaroonPermission, fullMethod string) (bool,
+	error) {
+
+	rpcPermissions := make([]*lnrpc.MacaroonPermission, len(permissions))
+	for _, perm := range permissions {
+		permission := &lnrpc.MacaroonPermission{
+			Entity: perm.Entity,
+			Action: perm.Action,
+		}
+
+		rpcPermissions = append(rpcPermissions, permission)
+	}
+
+	rpcCtx := s.adminMac.WithMacaroonAuth(ctx)
+	res, err := s.client.CheckMacaroonPermissions(
```

Comment:
This will only work once we pull in the version of `lnd` that has this new call. We can either wait until it's merged or temporarily add a `replace` directive to `go.mod`.

Reply:
Ah, replace it with this branch? https://github.com/lightningnetwork/lnd/pull/5304

Reply:
you can get the mod version using: 
```
$ go get github.com/orbitalturtle/lnd@{last commit on PR branch}
go: downloading github.com/orbitalturtle/lnd v0.0.1-beta 
```

Then this to the bottom of `go.mod`:
`replace github.com/lightningnetwork/lnd => github.com/orbitalturtle/lnd v0.0.1-beta`